src/jenkins.ts

```diff
@@ -0,0 +1,231 @@
+   *                { name: 'job-dsl', version: '1.78.1' },
+   *                { name: 'configuration-as-code', version: '1414.v878271fc496f' },
+   *                { name: 'kubernetes-credentials-provider', version: '0.20' }
+   */
+  readonly basePlugins?: Plugin[];
+  /**
+   * List of custom plugins applied to Jenkins resource.
+   *
+   * @default - []
+   */
+  readonly plugins?: Plugin[];
+  /**
+   * List of seed job configuration for Jenkins resource.
+   * For more information about seed jobs, please take a look at { @link https://github.com/jenkinsci/job-dsl-plugin/wiki/Tutorial---Using-the-Jenkins-Job-DSL Jenkins Seed Jobs Documentation }.
+   *
+   * @default - No seed jobs
+   */
+  readonly seedJobs?: SeedJob[];
+}
+
+/**
+ * A jenkins instance.
+ */
+export class Jenkins extends Construct {
+  private _plugins: Plugin[];
+  private _basePlugins: Plugin[];
+  private _seedJobs: SeedJob[];
+
+  constructor(scope: Construct, id: string, props: JenkinsProps = {}) {
+    super(scope, id);
+
+    this._plugins = props.plugins ?? [];
+    this._seedJobs = props.seedJobs ?? [];
+    this._basePlugins = [];
+    this.addBasePlugins(...(props.basePlugins ?? []));
+    const metadata = {
+      ...props.metadata,
+      namespace: props.metadata?.namespace ?? 'default',
+      labels: props.metadata?.labels ?? { app: 'jenkins' },
+    };
+    const authorizationStrategy = 'createUser';
+    const disableCSRFProtection = props.disableCsrfProtection ?? false;
+    // Keeping container resources defaults similar to Jenkins defaults here: https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/installing-the-operator/
+    const containers = [
+      {
+        name: 'jenkins-master',
+        image: 'jenkins/jenkins:lts',
+        imagePullPolicy: 'Always',
+        resources: {
+          limits: {
+            cpu: jenkins.JenkinsSpecMasterContainersResourcesLimits.fromString('1500m'),
+            memory: jenkins.JenkinsSpecMasterContainersResourcesLimits.fromString('3Gi'),
+          },
+          requests: {
+            cpu: jenkins.JenkinsSpecMasterContainersResourcesLimits.fromString('1'),
+            memory: jenkins.JenkinsSpecMasterContainersResourcesLimits.fromString('500M'),
+          },
+        },
+      },
+    ];;
+
+    const jenkinsCustomResource = new jenkins.Jenkins(this, 'Default', {
+      metadata: metadata,
+      spec: {
+        groovyScripts: {
+          configurations: [],
+          secret: {
+            name: '',
+          },
+        },
+        configurationAsCode: {
+          configurations: [],
+          secret: {
+            name: '',
+          },
+        },
+        jenkinsApiSettings: {
+          authorizationStrategy: authorizationStrategy,
+        },
+        master: {
+          disableCsrfProtection: disableCSRFProtection,
+          containers: containers,
+          basePlugins: cdk8s.Lazy.any({
+            produce: () => {
+              return this._basePlugins;
```

Comment:
`this._basePlugins` should be a map, and the conversion to an array should be placed here.

Reply:
Will update it.